src/provable/derived-leaves.ts

```diff
@@ -0,0 +1,135 @@
+import { GenericBool, GenericField, GenericHashInput } from './generic.js';
+import { createProvable } from './provable-generic.js';
+import * as Json from './gen/transaction-json.js';
+import { fieldEncodings, prefixToField } from './binable.js';
+
+export { derivedLeafTypes };
+
+function derivedLeafTypes<Field, Bool>({
```

Comment:
one of the many cases where we made an implementation generic over `Field`, so that we can reuse it between snarkyjs & mina-signer